qiskit_machine_learning/algorithms/effective_dimension.py

```diff
@@ -0,0 +1,337 @@
+        inputs: Optional[Union[List[float], np.ndarray, float]] = None,
+        num_params: Optional[int] = 1,
+        num_inputs: Optional[int] = 1,
+        fix_seed=False,
+        callback: Optional[Callable] = None,
+    ) -> None:
+        # pylint: disable=wrong-spelling-in-docstring
+        """
+        Args:
+            qnn: A Qiskit ``NeuralNetwork``, with a specific number
+                of weights/parameters (d = qnn_num_weights) that
+                will determine the shape of the Fisher Information Matrix
+                (num_inputs * num_params, d, d) used to compute the global
+                effective dimension for a set of ``inputs``, of shape
+                (num_inputs, qnn_input_size), and ``params``, of shape
+                (num_params, d).
+            params: An array of neural network parameters (weights), of shape
+                (num_params, qnn_num_weights).
+            inputs: An array of inputs to the neural network, of shape
+                (num_inputs, qnn_input_size).
+            num_params: If ``params`` is not provided, the algorithm will
+                randomly sample ``num_params`` parameter sets from a
+                uniform distribution. By default, num_params = 1.
+            num_inputs:  If ``inputs`` is not provided, the algorithm will
+                randomly sample ``num_inputs`` input sets
+                from a normal distribution. By default, num_inputs = 1.
+            callback: A callback function for the Montecarlo sampling.
+        """
+
+        # Store arguments
+        self._params = None
+        self._inputs = None
+        self._num_params = num_params
+        self._num_inputs = num_inputs
+        self._model = qnn
+        self._callback = callback
+
+        if fix_seed:
+            self._seed = 0
+        else:
+            self._seed = None
+        # Define inputs and parameters
+        self.params = params # type: ignore
+        # input setter uses self._model
+        self.inputs = inputs  # type: ignore
+
+    # keep d = num_weights for the sake of consistency with the
+    # nomenclature in the original code/paper
+    def d(self) -> int: # pylint: disable=invalid-name
+        """Returns the dimension of the model according to the definition
+        from the original paper."""
+        return self._model.num_weights
+
+    @property
+    def num_params(self) -> int:
+        """Returns the number of network parameter sets."""
+        return self._num_params
+
+    @num_params.setter
+    def num_params(self, num_params) -> None:
+        """Sets the number of network parameter sets."""
+        self._num_params = num_params
+
+    @property
+    def params(self) -> np.ndarray:
+        """Returns network parameters."""
+        return self._params
+
+    @params.setter
+    def params(self, params: Union[List[float], np.ndarray, float]) -> None:
+        """Sets network parameters."""
+        if params is not None:
```

Comment:
If params is allowed to be None here then the typehint should have Optional i.e. `Optional[Union[List[float], np.ndarray, float])]`